app/models/referral.rb

```diff
@@ -0,0 +1,58 @@
+require "google_drive"
+
+class Referral
+  include ActiveModel::Model
+
+  def self.worksheet
+    service_account_io = OpenStruct.new(read: "{
+      \"type\": \"service_account\",
+      \"project_id\": \"#{ENV["GOOGLE_DRIVE_PROJECT_ID"]}\",
```

Comment:
Not essential but consider using `ENV#fetch` here for more predictable explosions if there are missing ENV variables.